src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneDebitMethodSender.php

```diff
@@ -0,0 +1,117 @@
+use SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader;
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class PayoneDebitMethodSender extends AbstractPayoneMethodSender implements PayoneDebitMethodSenderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface
+     */
+    protected $executionAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @var \Generated\Shared\Transfer\PayoneStandardParameterTransfer
+     */
+    protected $standardParameter;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface
+     */
+    protected $standartParameterMapper;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Business\Api\Adapter\AdapterInterface $executionAdapter
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PayoneStandardParameterTransfer $standardParameter,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PaymentMapperReader $paymentMapperReader
+    ) {
+        parent::__construct($queryContainer, $paymentMapperReader);
+        $this->executionAdapter = $executionAdapter;
+        $this->queryContainer = $queryContainer;
+        $this->standardParameter = $standardParameter;
+        $this->standartParameterMapper = $standartParameterMapper;
+    }
+
+    /**
+     * @param int $idPayment
+     *
+     * @return \Generated\Shared\Transfer\DebitResponseTransfer
+     */
+    public function debitPayment(int $idPayment): DebitResponseTransfer
+    {
+        $paymentEntity = $this->getPaymentEntity($idPayment);
+        $paymentMethodMapper = $this->getPaymentMethodMapper($paymentEntity);
+
+        $requestContainer = $paymentMethodMapper->mapPaymentToDebit($paymentEntity);
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+
+        $paymentEntity = $this->findPaymentByTransactionId($paymentEntity->getTransactionId());
+        $apiLogEntity = $this->initializeApiLog($paymentEntity, $requestContainer);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+        $responseContainer = new DebitResponseContainer($rawResponse);
+
+        $this->updateApiLogAfterDebit($apiLogEntity, $responseContainer);
+
+        $responseMapper = new DebitResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
++